### PR TITLE
[six][docs] adds doxygen style documentation to six + minor cleanup

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

Adds doxygen style docs to `six`.

### Motivation

Improved documentation for the future with relevant python, memory management comments, etc.

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: six/six/api.cpp [hunk 1]
```diff
@@ -68,8 +81,16 @@ create_t *loadAndCreate(const char *dll, const char *python_home, char **error)
     }
     return create;
 }
+
 six_t *make2(const char *python_home, char **error)
 {
+
+    if (six_backend != NULL) {
+        std::string err_msg = "Six already initialized!";
+        *error = strdup(err_msg.c_str());
```
Comment (derekwbrown) [inline six/six/api.cpp]: it's an error case, so it's not a big deal (and not likely to be hit), but this is inefficient.  Creates a `std::string` on the stack, allocates memory to store "six already...", and then just re-allocates the memory for return.

*error = strdup("Six already initialized") would be fine.

File: six/six/api.cpp [hunk 2]
```diff
@@ -79,13 +100,24 @@ six_t *make2(const char *python_home, char **error)
 
 six_t *make3(const char *python_home, char **error)
 {
+    if (six_backend != NULL) {
+        std::string err_msg = "Six already initialized!";
```
Comment (derekwbrown) [inline six/six/api.cpp]: Same.

**Review by derekwbrown — DISMISSED** [review]
Very minor nits; fine after updates (if you even think they're worth it)

**Review by derekwbrown — DISMISSED** [review]

**Review by albertvaka — DISMISSED** [review]

File: six/common/builtins/aggregator.c [hunk 3]
```diff
@@ -23,6 +23,16 @@ static PyMethodDef methods[] = {
     { NULL, NULL } // guards
 };
 
+/*! \fn add_constants(PyObject *m)
+    \brief A helper function to add a a set of constants to a python module.
+    \param m A PyObject * pointer to  the module you wish to add the constant to.
+    \return a char ** pointer to the C-representation of the provided python
```
Comment (albertvaka) [inline six/common/builtins/aggregator.c]: Doesn't return.

File: six/common/builtins/tagger.c [hunk 4]
```diff
@@ -36,15 +60,28 @@ PyObject *buildTagsList(char **tags)
...
     }
     cgo_free(tags);
     return res;
 }
 
+/*! \fn PyObject *tag(PyObject *self, PyObject *args)
+    \brief builds a tag list as per the entity id and cardinality passed as method
+    arguments.
+    \param self A PyObject* pointer to the tagger module.
+    \param args A PyObject* pointer to the tag method, typically expected to
```
Comment (albertvaka) [inline six/common/builtins/tagger.c]:
```suggestion
    \param args A PyObject* pointer to the python args, expected to
```

File: six/common/builtins/tagger.c [hunk 5]
```diff
@@ -36,15 +60,28 @@ PyObject *buildTagsList(char **tags)
...
+    contain the id and the cardinality..
+    \return a PyObject * pointer to the python tag list.
```
Comment (albertvaka) [inline six/common/builtins/tagger.c]:
```suggestion
    \return a PyObject * pointer to the tag list, NONE if callback not set, or NULL in an error.
```

File: six/common/builtins/tagger.c [hunk 6]
```diff
@@ -72,6 +109,21 @@ PyObject *tag(PyObject *self, PyObject *args)
     return buildTagsList(cb_tags(id, cardinality));
 }
 
+/*! \fn PyObject *get_tag(PyObject *self, PyObject *args)
+    \brief builds a tag list as per the entity id and cardinality passed as
+    arguments.
+    \param self A PyObject* pointer to the tagger module.
+    \param args A PyObject* pointer to the python args, typically expected to
```
Comment (albertvaka) [inline six/common/builtins/tagger.c]:
```suggestion
    \param args A PyObject* pointer to the python args, expected to
```

File: six/common/builtins/util.h [hunk 7]
```diff
@@ -4,20 +4,40 @@
...
+    array in 'util.c' and registered into python. This function is python3 only.
+*/
+PyMODINIT_FUNC PyInit_util(void);
+#elif defined(DATADOG_AGENT_TWO)
+
+/*! \fn void Py2_init_util()
+    \brief Initializes the util builtin python module.
+
+    The 'util' python builtin is created with the methods from the PyMethodDef
+    array in 'util.c' and registered into python. This function is python3 only.
```
Comment (albertvaka) [inline six/common/builtins/util.h]:
```suggestion
    array in 'util.c' and registered into python. This function is python2 only.
```

File: six/six/api.cpp [hunk 8]
```diff
@@ -40,6 +40,19 @@ static void *six_backend = NULL;
 
 #ifdef _WIN32
 
+/*! \fn create_t *loadAndCreate(const char *dll, const char *python_home, char **error)
+    \brief Loads the Python backend DLL from the provided PYTHONHOME, and returns its
+    creation routine.
+    \param dll A C-string containing the expected backend DLL name.
+    \param python_home A C-string containing the expected PYTHONhOME for said DLL.
+    \param error A C-stringi pointer output parameter to return error messages.
```
Comment (albertvaka) [inline six/six/api.cpp]:
```suggestion
    \param error A C-string pointer output parameter to return error messages.
```

File: six/six/api.cpp [hunk 9]
```diff
@@ -40,6 +40,19 @@ static void *six_backend = NULL;
 
 #ifdef _WIN32
 
+/*! \fn create_t *loadAndCreate(const char *dll, const char *python_home, char **error)
+    \brief Loads the Python backend DLL from the provided PYTHONHOME, and returns its
+    creation routine.
+    \param dll A C-string containing the expected backend DLL name.
+    \param python_home A C-string containing the expected PYTHONhOME for said DLL.
```
Comment (albertvaka) [inline six/six/api.cpp]:
```suggestion
    \param python_home A C-string containing the expected PYTHONHOME for said DLL.
```

File: six/three/three.h [hunk 10]
```diff
@@ -84,17 +111,57 @@ class Three : public Six
     void setIsExcludedCb(cb_is_excluded_t);
 
 private:
+    //! initPythonHome member.
+    /*!
+      \brief This member function sets the Python home for the underlying python2.7 interpreter.
```
Comment (albertvaka) [inline six/three/three.h]:
```suggestion
      \brief This member function sets the Python home for the underlying python3 interpreter.
```

File: six/include/six.h [hunk 11]
```diff
@@ -12,91 +12,387 @@
...
+  The goal of this class is to provide a simple, comprehensive interface to allow
+  implementing runtimes to integrate with the agent and run integration checks.
+*/
 class Six
 {
 public:
+    //! Constructor.
+    /*!
+      Basic constructor, initializes the _error string to an empty string and
+      errorFlag to false.
```
[the remark this answers is not in the document]
Comment (truthbk) [reply six/include/six.h]: While I do agree for the general case I find it sometimes useful to find the constructor defaults documented when the constructor takes no arguments. Anyway, the attributes are pretty non-consequential to the behavior so I'll remove the comment.

**Review by albertvaka — APPROVED** [review]
🥇

**Pull request merged**